x-pack/plugins/beats/server/routes/api/register_enroll_beat_route.js

```diff
@@ -25,7 +24,14 @@ async function getEnrollmentToken(callWithInternalUser, enrollmentToken) {
   };
 
   const response = await callWithInternalUser('get', params);
-  return get(response, '_source.enrollment_token', {});
+  const token = get(response, '_source.enrollment_token', {});
+
+  // Elasticsearch might return fast if the token is not found. OR it might return fast
+  // if the token *is* found. Either way, an attacker could using a timing attack to figure
+  // out whether a token is valid or not. So we introduce a random delay in returning from
+  // this function to obscure the actual time it took for Elasticsearch to find the token.
+  const randomDelayInMs = 25 + Math.round(Math.random() * 200); // between 25 and 225 ms
```

Comment:
See comment re. clamping. That's a lot harder to de-noise than this, see https://blog.ircmaxell.com/2014/11/its-all-about-time.html

This approach is basically spewing random numbers that even makes it reasonably easy to find the seed similar to this - https://franklinta.com/2014/08/31/predicting-the-next-math-random-in-java/

I don't think this timing sidechannel is a big deal (if the key isn't valid, it doesn't exist at this point?), but if we're going to remove it, I don't think random noise is the way :)

Reply:
Yes, I'm starting to think protecting against a timing attack just to determine existence of the key is probably overkill in this case. That's because a) these enrollment tokens are meant to be short-lived and one-time use and b) they have an expiration time as well. In other words, by the time a timing attack might determine an enrollment token, it might've been used or have expired already. So now I'm inclined to remove the random noise bit I implemented in 7af9c2d.

@alexbrasetvik sounds like you're on board with that. @kobelb WDYT?

Reply:
I'd agree.

https://events.ccc.de/congress/2012/Fahrplan/attachments/2235_29c3-schinzel.pdf is also a good resource on this topic.

Reply:
Sounds good to me!